pkg/solana/chainwriter/transform_registry.go

```diff
@@ -316,7 +415,154 @@ func appendTokenTransferAccounts(tokenAccountsRequired bool, accounts solana.Acc
 	return accounts, tokenIndexes, nil
 }
 
-func getFeeQuoterAddress(ctx context.Context, toAddress string, args any, tableMap map[string]map[string][]*solana.AccountMeta, client client.MultiClient) (solana.PublicKey, error) {
+func deriveExecuteAccounts(ctx context.Context, client client.MultiClient, params ccip_offramp.DeriveAccountsExecuteParams, messageTokenData [][]byte, transmitter solana.PublicKey, offrampStr string, lggr logger.Logger) (solana.AccountMetaSlice, map[solana.PublicKey]solana.PublicKeySlice, []uint8, error) {
+	blockhash, err := client.LatestBlockhash(ctx)
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("error fetching latest blockhash: %w", err)
+	}
+	offramp, err := solana.PublicKeyFromBase58(offrampStr)
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("failed to parse offramp address: %w", err)
+	}
+	config, _, err := state.FindOfframpConfigPDA(offramp)
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("failed to calculate offramp config address: %w", err)
+	}
+	var derivedAccounts, accountsToAskWith solana.AccountMetaSlice
+	lookupTableMap := make(map[solana.PublicKey]solana.PublicKeySlice)
+	tokenIndexes := []uint8{}
+	mandatoryAccountsLen := cap(ccip_offramp.NewExecuteInstructionBuilder().AccountMetaSlice)
+	stage := "Start"
+	ttAccountsMatcher, err := regexp.Compile(`^TokenTransferStaticAccounts/\d+/0$`)
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("failed to compile token transfer stage matcher: %w", err)
+	}
+	for {
+		deriveAccountsIxRaw := ccip_offramp.NewDeriveAccountsExecuteInstruction(params, stage, config)
+		deriveAccountsIxRaw.AccountMetaSlice = append(deriveAccountsIxRaw.AccountMetaSlice, accountsToAskWith...)
+		deriveAccountsIx, err := deriveAccountsIxRaw.ValidateAndBuild()
+		if err != nil {
+			return nil, nil, nil, fmt.Errorf("failed to build derive execute accounts instruction: %w", err)
+		}
+		deriveAccountsIxData, err := deriveAccountsIx.Data()
+		if err != nil {
+			return nil, nil, nil, fmt.Errorf("failed to encode account derivation instruction data: %w", err)
+		}
+		deriveAccountsSolIx := solana.NewInstruction(offramp, deriveAccountsIx.Accounts(), deriveAccountsIxData)
```

Comment:
Similar to a previous question, is there a reason to override the program id of an instruction meant for the offramp with the offramp ID again? There's no negative impact from this anyway other than extra lines of code, I'm just curious 👀

Reply:
Ya it was the same issue as above